modules/top-level/test.nix

```diff
@@ -19,23 +90,79 @@ in
       description = "The test derivation's name.";
     };
 
+    buildNixvim = lib.mkOption {
+      type = lib.types.bool;
+      description = "Whether to build the nixvim config in the test.";
+      default = true;
+    };
+
     runNvim = lib.mkOption {
       type = lib.types.bool;
       description = "Whether to run `nvim` in the test.";
-      default = true;
+      defaultText = lib.literalExpression "config.test.buildNixvim";
+      default = cfg.buildNixvim;
     };
 
     checkWarnings = lib.mkOption {
       type = lib.types.bool;
       description = "Whether to check `config.warnings` in the test.";
+      apply = x: lib.warnIfNot x "`test.checkWarnings = false` is replaced with `test.warnings = [ ]`." x;
       default = true;
     };
 
     checkAssertions = lib.mkOption {
       type = lib.types.bool;
       description = "Whether to check `config.assertions` in the test.";
+      apply =
+        x: lib.warnIfNot x "`test.checkAssertions = false` is replaced with `test.assertions = [ ]`." x;
       default = true;
     };
+
+    warnings = lib.mkOption {
+      type = lib.types.listOf expectationType;
+      description = "A list of expectations for `warnings`.";
+      defaultText = lib.literalMD "Expect `count == 0`";
+      default = [ 0 ];
```

Comment:
Why isn't the default merged with the values set by the test?
In particular why doesn't it result in : `[ { expect = "count"; value = 0;} { expect = "count"; value = 1;} { expect = "any", value ="The rust-tools project has been abandoned."; } ]` in one of the tests?

Reply:
Only definitions of the same override-priority get merged together.

The default is prio 1500 (mkOptionDefault) while normal definitions are prio 100 (`lib.modules.defaultOverridePriority`)

This is intentional; the default list asserts that no warnings are present, but if you manually define expectations we don't want this.

Reply:
Yeah I saw the intention, but I did not know enough of the module system to understand how it works